### PR TITLE
Add Barlett's test

### PR DESCRIPTION
**Pull request by ararslan**
Conflicts with #146 and more significantly with #145.

Imported from https://github.com/ararslan/MultivariateTests.jl, test values validated against SAS when originally written.

### CODE REVIEW

File: src/bartlett.jl [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+# Finite population correction factor
+@inline _correction(p::Int, nxm1::Int, nym1::Int) =
+    1 - ((2p^2 + 3p - 1) * (inv(nxm1) + inv(nym1) - inv(nxm1 + nym1)) / 6(p + 1))
+
+# Pool the given covariance matrices, trashing them both
+function _poolcov!(Sx::AbstractMatrix, nxm1::Int, Sy::AbstractMatrix, nym1::Int)
+    Sx .*= nxm1
+    Sy .*= nym1
+    Sx .+= Sy
```
Comment (nalimilan) [inline src/bartlett.jl]: If you compute `Sy` first, you can update `Sx` in a single line, which will allow for loop fusion.

Comment (ararslan) [reply src/bartlett.jl]: Very nice!

Comment (ararslan) [reply src/bartlett.jl]: Actually, this is the same function as in #145, so once that's merged I'll just remove this entirely, as the version in #145 is faster than what's here.

File: src/bartlett.jl [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+## Utility functions
+
+# Finite population correction factor
+@inline _correction(p::Int, nxm1::Int, nym1::Int) =
+    1 - ((2p^2 + 3p - 1) * (inv(nxm1) + inv(nym1) - inv(nxm1 + nym1)) / 6(p + 1))
+
+## Bartlett's test
+
+struct BartlettsTest <: CovarianceEqualityTest
+    L′::Real
```
[the remark this answers is not in the document]
Comment (ararslan) [reply src/bartlett.jl]: That's fair. I was just using the notation in the reference I went from. It won't be user-facing if we do #79. 🙂

[the remark this answers is not in the document]
Comment (mschauer) [reply src/bartlett.jl]: Ah, great. :-)

[the remark this answers is not in the document]
Comment (nalimilan) [reply src/bartlett.jl]: More importantly, shouldn't this be a type parameter?

[the remark this answers is not in the document]
Comment (mschauer) [reply src/bartlett.jl]: Ah, it is a `Real`. But it is somewhat hard to tell what type `L'` will be in `function BartlettsTest(X::AbstractMatrix, Y::AbstractMatrix)`

**Pull request merged**